src/views/TradingCompetition/index.tsx

```diff
@@ -201,8 +204,12 @@ const TradingCompetition = () => {
   }, [profileApiUrl])
 
   // Don't hide when loading. Hide if the account is connected && the user hasn't registered && the competition is live or finished
-  const shouldHideCta =
-    !isLoading && account && !userTradingInformation.hasRegistered && (isCompetitionLive || hasCompetitionEnded)
+  const shouldHideCta = [
+    !isLoading,
+    account,
+    !userTradingInformation.hasRegistered,
+    isCompetitionLive || hasCompetitionEnded
+  ].every(i => Boolean(i))
```

Comment:
I am not convinced this is more _readable_. I would opt for an inline function.
```
const shouldHideCta = () => {
  if (!isLoading) {
    return false
  }  

// Guard for each item  
...etc
  
  return true
}
```

Reply:
I got it, this case can be more controllable.
But now the constance "shouldHideCta" type will become to Function.

Reply:
Updated.

d68109852c96cc46d309396f0e54c258c9274505